memory_engine/memory_store.py

```
@@ -50,165 +50,174 @@ def __init__(self, decay_lambda: float = 0.1, max_capacity: int = 10000):
         os.makedirs(os.path.dirname(self.storage_path), exist_ok=True)
-        with open(self.storage_path, 'wb') as f:
+        with open(self.storage_path, "wb") as f:
             pickle.dump(self.memory, f)
-    
+
     def load(self) -> bool:
         """Load memory from disk."""
         if os.path.exists(self.storage_path):
-            with open(self.storage_path, 'rb') as f:
+            with open(self.storage_path, "rb") as f:
                 self.memory = pickle.load(f)
```

Ruff (0.14.10)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)